src/shainet/basic/exprimental.cr

```diff
@@ -39,7 +39,7 @@ module SHAInet
       end
 
       unless stealth # Hide report during training
-        @input_layers.last.activations.show(@logger, "input layer activations:")
+        @input_layers.last.activations.show(Log, "input layer activations:")
```

Comment:
As I said before, passing a `Log` instance shouldn't be needed.
ditto elsewhere